### PR TITLE
support priority processing claim tx

### PR DESCRIPTION
**Pull request by chengzhinei**

### CODE REVIEW

File: cmd/txpool/main.go [hunk 1]
```diff
@@ -176,6 +180,15 @@ func doTxpool(ctx context.Context) error {
 		addr := common.HexToAddress(addrHex)
 		ethCfg.DeprecatedTxPool.BlockedList[i] = addr.String()
 	}
+	ethCfg.DeprecatedTxPool.FreeClaimGasAddr = make([]string, len(freeClaimGasAddr))
+	for i, addrHex := range freeClaimGasAddr {
+		addr := common.HexToAddress(addrHex)
+		ethCfg.DeprecatedTxPool.FreeClaimGasAddr[i] = addr.String()
```
Comment (giskook) [inline cmd/txpool/main.go:186]: Is there some reason why they named `DeprecatedTxPool`? is there any other configure to used by txpool?

**Review by giskook — APPROVED** [review]

File: cmd/txpool/main.go [hunk 2]
```diff
@@ -83,6 +85,8 @@ func init() {
 	rootCmd.PersistentFlags().Uint64Var(&priceBump, "txpool.pricebump", txpoolcfg.DefaultConfig.PriceBump, "Price bump percentage to replace an already existing transaction")
 	rootCmd.PersistentFlags().DurationVar(&commitEvery, utils.TxPoolCommitEveryFlag.Name, utils.TxPoolCommitEveryFlag.Value, utils.TxPoolCommitEveryFlag.Usage)
 	rootCmd.Flags().StringSliceVar(&traceSenders, utils.TxPoolTraceSendersFlag.Name, []string{}, utils.TxPoolTraceSendersFlag.Usage)
+	rootCmd.Flags().StringSliceVar(&freeClaimGasAddr, utils.TxPoolPackBatchSpacialList.Name, []string{"0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266"}, utils.TxPoolPackBatchSpacialList.Usage)
```
Comment (sieniven) [inline cmd/txpool/main.go:88]:
```suggestion
	// For X Layer
	rootCmd.Flags().StringSliceVar(&freeClaimGasAddr, utils.TxPoolPackBatchSpecialList.Name, []string{"0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266"}, utils.TxPoolPackBatchSpecialList.Usage)
```

File: cmd/utils/flags_xlayer.go [hunk 3]
```diff
@@ -67,6 +67,16 @@ var (
 		Usage: "Comma separated list of addresses, who can't send and receive transactions",
 		Value: "",
 	}
+	TxPoolPackBatchSpacialList = cli.StringFlag{
+		Name:  "txpool.packbatchspaciallist",
```
Comment (sieniven) [inline cmd/utils/flags_xlayer.go:71]:
```suggestion
	TxPoolPackBatchSpecialList = cli.StringFlag{
		Name:  "txpool.packbatchspeciallist",
```

File: cmd/utils/flags_xlayer.go [hunk 4]
```diff
@@ -240,4 +250,19 @@ func setTxPoolXLayer(ctx *cli.Context, cfg *ethconfig.DeprecatedTxPoolConfig) {
...
+		addrHexes := SplitAndTrim(ctx.String(TxPoolPackBatchSpacialList.Name))
+
+		cfg.FreeClaimGasAddr = make([]string, len(addrHexes))
+		for i, senderHex := range addrHexes {
+			sender := libcommon.HexToAddress(senderHex)
+			cfg.FreeClaimGasAddr[i] = sender.String()
+		}
+		if len(cfg.FreeClaimGasAddr) == 0 {
+			cfg.FreeClaimGasAddr = []string{"0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266"}
+		}
```
Comment (sieniven) [inline cmd/utils/flags_xlayer.go:263]: Will we ever go into this pipeline? Should this be added when the TxPoolPackBatchSpecialList is not set?

File: test/config/test.erigon.seq.config.yaml [hunk 5]
```diff
@@ -64,6 +64,8 @@ db.read.concurrency: 20000
 txpool.globalslots: 100000
 txpool.globalbasefeeslots: 100000
 txpool.globalqueue: 100000
+txpool.packbatchspaciallist : ["0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266"]
```
Comment (sieniven) [inline test/config/test.erigon.seq.config.yaml:67]:
```suggestion
txpool.packbatchspeciallist : ["0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266"]
```

File: turbo/cli/default_flags.go [hunk 6]
```diff
@@ -26,6 +26,8 @@ var DefaultFlags = []cli.Flag{
 	&utils.TxPoolLifetimeFlag,
 	&utils.TxPoolTraceSendersFlag,
 	&utils.TxPoolCommitEveryFlag,
+	&utils.TxPoolPackBatchSpacialList,
```
Comment (sieniven) [inline turbo/cli/default_flags.go:29]:
```suggestion
	&utils.TxPoolPackBatchSpecialList,
```

File: zk/txpool/txpooluitl/all_components.go [hunk 7]
```diff
@@ -22,8 +22,8 @@ import (
 	"time"
 
 	"github.com/c2h5oh/datasize"
-	"github.com/holiman/uint256"
 	"github.com/gateway-fm/cdk-erigon-lib/txpool/txpoolcfg"
+	"github.com/holiman/uint256"
```
Comment (sieniven) [inline zk/txpool/txpooluitl/all_components.go:26]: Revert file change

Comment (chengzhinei) [reply zk/txpool/txpooluitl/all_components.go:26]: this is goland fmt auto changed. Maybe should use right fmt

File: cmd/utils/flags_xlayer.go (hunk 4 above)
Comment (chengzhinei) [reply cmd/utils/flags_xlayer.go:263]: default address of bridge-server should be added. keep same with zknode

Comment (sieniven) [reply cmd/utils/flags_xlayer.go:263]: Dont understand how this resolves what I was asking. If your config "txpool.packbatchspeciallist" is set, then it is very likely you are specifying a list of free claim gas addresses. You should be taking the codeblock for the check "if len(cfg.FreeClaimGasAddr) == 0" out of the scope when your config "txpool.packbatchspeciallist" is set.

**Review by sieniven — APPROVED** [review]

File: cmd/utils/flags_xlayer.go [hunk 8]
```diff
@@ -240,4 +250,19 @@ func setTxPoolXLayer(ctx *cli.Context, cfg *ethconfig.DeprecatedTxPoolConfig) {
...
+	if ctx.IsSet(TxPoolPackBatchSpacialList.Name) {
+		addrHexes := SplitAndTrim(ctx.String(TxPoolPackBatchSpacialList.Name))
+
+		cfg.FreeClaimGasAddr = make([]string, len(addrHexes))
+		for i, senderHex := range addrHexes {
+			sender := libcommon.HexToAddress(senderHex)
+			cfg.FreeClaimGasAddr[i] = sender.String()
+		}
+		if len(cfg.FreeClaimGasAddr) == 0 {
+			cfg.FreeClaimGasAddr = []string{"0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266"}
```
Comment (zjg555543) [inline cmd/utils/flags_xlayer.go]: 0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266？

Comment (zjg555543) [reply cmd/utils/flags_xlayer.go]: zkevm.address-sequencer can be used.

**Pull request merged**